### PR TITLE
refactor: rename ArrayInfo to VariableInfo across the codebase

### LINKED ISSUE #182: refactor(netcdf): rename ArrayInfo to VariableInfo and arrays to variables
## Context

The GDAL multidimensional API uses the term "array" (MDArray), but NetCDF's standard
terminology is "variable". This mismatch confuses users who expect NetCDF vocabulary.

## Problem / Current Behaviour

`ArrayInfo`, `arrays` field on `GroupInfo` and `NetCDFMetadata`, and all internal
references use "array" terminology.

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/models.py` | `ArrayInfo`, `GroupInfo.arrays`, `NetCDFMetadata.arrays` | class + fields |
| `src/pyramids/netcdf/metadata.py` | `GroupTraverser.arrays`, `MetadataBuilder.build()` | builder + traverser |
| `src/pyramids/netcdf/metadata.py` | `to_dict`, `to_json`, `from_json`, `flatten_for_index` | serialization |
| `src/pyramids/netcdf/__init__.py` | `__all__` exports | public API |

## Proposed Solution

- Rename `ArrayInfo` → `VariableInfo`
- Rename `arrays` → `variables` on `GroupInfo`, `NetCDFMetadata`, `GroupTraverser`,
  `MetadataBuilder`, and all serialization functions
- Update docstrings, doctests, and all test files

## Out of Scope

- Renaming internal GDAL references (`OpenMDArray`, `GetMDArrayNames`)

## Effort Estimate

**Size**: `M`
**Rationale**: mechanical rename across ~10 files plus test updates

## Definition of Done

- [ ] `ArrayInfo` no longer exists; `VariableInfo` is the public name
- [ ] All serialization keys use `variables` not `arrays`
- [ ] All existing tests pass with updated names
- [ ] Docstrings/doctests updated

### LINKED ISSUE #183: feat(netcdf): add sel() for coordinate-based band selection
## Context

Users working with multi-temporal NetCDF data need to select bands by coordinate
values (e.g. time steps) rather than band indices, similar to xarray's `.sel()`.

## Problem / Current Behaviour

No way to select bands by coordinate values. Users must manually find band indices
matching their desired coordinate values.

## Proposed Solution

Add `sel(**kwargs)` to `NetCDF` that accepts:
- A single value: `var.sel(time=6)` — select one band
- A list: `var.sel(time=[0, 12, 24])` — select multiple bands
- A slice: `var.sel(time=slice(6, 18))` — select a range

Works on variable subsets that have `_band_dim_name` and `_band_dim_values`
set by `get_variable()`.

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.sel` | new method |

## Out of Scope

- Multi-dimension selection (only one keyword arg supported)
- Nearest-neighbor matching (exact match only)

## Effort Estimate

**Size**: `M`
**Rationale**: new method + comprehensive test suite (test_sel.py, ~460 lines)

## Definition of Done

- [ ] `sel()` works with single value, list, and slice selectors
- [ ] Raises `ValueError` for mismatched dimension names or no matches
- [ ] Returns a new Dataset with selected bands and updated coordinates
- [ ] Tests cover all selector types and edge cases

### LINKED ISSUE #184: feat(netcdf): enable crop/to_crs/resample on root MDIM containers
## Context

Spatial operations (crop, reproject, resample) on a root MDIM container currently
raise ValueError, forcing users to manually loop over variables.

## Problem / Current Behaviour

```python
nc = NetCDF.read_file("multi_var.nc")
nc.crop(mask)  # raises ValueError: "crop is not supported on MDIM containers"
```

## Proposed Solution

When `crop()`, `to_crs()`, or `resample()` are called on a root MDIM container
(detected via `_is_md_array and not _is_subset and band_count == 0`), apply the
operation to every variable via `_apply_to_all_variables()` and return a new
NetCDF container with the results.

On variable subsets, delegate to parent `Dataset` methods as before.

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.crop` | container mode |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.to_crs` | container mode |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.resample` | new override |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF._apply_to_all_variables` | new helper |

## Out of Scope

- `align` and `mosaic` on containers

## Effort Estimate

**Size**: `M`
**Rationale**: one shared helper + three method overrides + tests

## Definition of Done

- [ ] crop/to_crs/resample work on root containers with multiple variables
- [ ] Returns a new container (original unchanged)
- [ ] Band counts preserved across all variables
- [ ] Tests for each operation on multi-variable containers

### LINKED ISSUE #185: feat(netcdf): add unpack parameter to read_array for CF scale/offset
## Context

CF-convention NetCDF files often store packed data with `scale_factor` and
`add_offset` attributes. Users need to unpack to physical values.

## Problem / Current Behaviour

`read_array()` returns raw packed values. Users must manually apply the
transformation `real = raw * scale + offset`.

## Proposed Solution

Add `unpack: bool = False` parameter to `NetCDF.read_array()`. When True,
applies `result = result * scale + offset` using `_scale` and `_offset`
attributes stored on the variable subset by `get_variable()`.

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.read_array` | add unpack param |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.get_variable` | store _scale/_offset |

## Out of Scope

- Automatic unpacking (always opt-in via `unpack=True`)
- Missing value masking

## Effort Estimate

**Size**: `S`
**Rationale**: small addition to read_array + get_variable, new test file

## Definition of Done

- [ ] `read_array(unpack=True)` applies scale/offset
- [ ] `read_array(unpack=False)` returns raw data (default, backward compat)
- [ ] get_variable stores _scale and _offset from MDArray
- [ ] Tests cover scale-only, offset-only, both, and neither cases

### LINKED ISSUE #186: refactor(netcdf): lazy-load variables via _LazyVariableDict
## Context

The `variables` property eagerly loads every variable via `get_variable()`
on first access, which involves `AsClassicDataset` + Y-flip per variable.
This is expensive for files with many variables.

## Problem / Current Behaviour

```python
nc.variables  # triggers get_variable() for ALL variables upfront
```

## Proposed Solution

Replace the eager dict comprehension with `_LazyVariableDict` that only
calls `get_variable()` when a specific key is first accessed. Implements
full dict interface: `__getitem__`, `get`, `__contains__`, `__len__`,
`__iter__`, `keys`, `values`, `items`.

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `_LazyVariableDict` | new class |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.variables` | use lazy dict |

## Effort Estimate

**Size**: `S`
**Rationale**: one new class, one property change

## Definition of Done

- [ ] `variables` property returns a `_LazyVariableDict`
- [ ] Variables only loaded on first access per key
- [ ] Full dict interface works (iteration, containment, len)
- [ ] Cache invalidation still works via `_invalidate_caches`

### LINKED ISSUE #187: refactor(netcdf): add __str__/__repr__ to NetCDFMetadata, remove names and dimension_overview
## Context

`NetCDFMetadata` lacks human-readable display methods and carries redundant
fields (`names` property, `dimension_overview`) that duplicate information
already available in the `dimensions` dict.

## Problem / Current Behaviour

- `print(metadata)` shows raw dataclass repr with all fields
- `names` property duplicates `list(meta.dimensions.keys())`
- `dimension_overview` and `_build_dimension_overview()` add complexity
  without unique value
- `get_all_metadata()` exists only to populate `dimension_overview`

## Proposed Solution

- Add `__str__` with formatted summary (dimensions, variables, attributes)
- Add `__repr__` with counts and driver info
- Remove `names` property (use `list(meta.dimensions.keys())` directly)
- Remove `dimension_overview` field and `_build_dimension_overview()` method
- Simplify `get_all_metadata()` to just call `get_metadata()` without post-processing

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/models.py` | `NetCDFMetadata` | add str/repr, remove fields |
| `src/pyramids/netcdf/netcdf.py` | `get_all_metadata`, `_build_dimension_overview` | simplify/remove |

## Effort Estimate

**Size**: `S`
**Rationale**: mostly deletion + two new dunder methods

## Definition of Done

- [ ] `__str__` shows readable summary with dimensions, variables, attributes
- [ ] `__repr__` shows compact technical representation
- [ ] `names` property removed
- [ ] `dimension_overview` field and builder removed
- [ ] All tests updated

### LINKED ISSUE #188: fix(netcdf): strip leading slash from dimension/variable/group keys
## Context

Metadata dictionary keys include a leading `/` (e.g. `/time`, `/temperature`)
which is unintuitive and inconsistent with how users reference dimensions
and variables by name.

## Problem / Current Behaviour

```python
meta.dimensions  # {"/time": DimensionInfo(...), "/lat": ...}
meta.variables   # {"/temperature": VariableInfo(...)}
meta.groups      # {"/": GroupInfo(...), "/forecast": ...}
```

## Proposed Solution

Strip the leading `/` from keys when populating the dictionaries in
`GroupTraverser._collect_dimensions`, `_collect_arrays`, and `walk`.
Exception: root group key stays as `"/"`.

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/metadata.py` | `GroupTraverser._collect_dimensions` | lstrip("/") |
| `src/pyramids/netcdf/metadata.py` | `GroupTraverser._collect_arrays` | lstrip("/") |
| `src/pyramids/netcdf/metadata.py` | `GroupTraverser.walk` | lstrip("/") except "/" |

## Effort Estimate

**Size**: `XS`
**Rationale**: three lines changed in traverser + test updates

## Definition of Done

- [ ] Dimension keys are `"time"` not `"/time"`
- [ ] Variable keys are `"temperature"` not `"/temperature"`
- [ ] Root group key remains `"/"`
- [ ] `get_dimension()` still works with both short and full names

### LINKED ISSUE #189: feat(netcdf): add chunking and compression support to create_from_array
## Context

Users creating NetCDF files need control over chunking and compression for
performance and file size optimization, especially for large datasets.

## Problem / Current Behaviour

`create_from_array` always creates unchunked, uncompressed NetCDF files.
No way to specify chunk sizes or compression algorithm.

## Proposed Solution

Add three parameters to `create_from_array` and `_build_mdim_dataset`:
- `chunk_sizes: tuple | list | None` — chunk dimensions (e.g. `(1, 256, 256)`)
- `compression: str | None` — algorithm name (`"DEFLATE"`, `"ZSTD"`, etc.)
- `compression_level: int | None` — compression level (e.g. 1-9 for DEFLATE)

Also:
- Add `_create_dimension` helper that supports `set_indexing=True/False`
  (netCDF driver doesn't support `SetIndexingVariable`)
- Fix nodata-before-write ordering (netCDF driver requires `SetNoDataValueDouble`
  before first `Write` call)
- Creation options only apply when writing to disk (`path` is not None)

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.create_from_array` | new params |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF._build_mdim_dataset` | impl |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF._create_dimension` | new helper |

## Effort Estimate

**Size**: `L`
**Rationale**: driver-specific logic + comprehensive test file (390 lines)

## Definition of Done

- [ ] DEFLATE compression reduces file size
- [ ] Chunk sizes readable via GDAL MDIM API
- [ ] Chunking + compression work together
- [ ] Data round-trips correctly through compressed files
- [ ] Nodata preserved on disk
- [ ] In-memory creation ignores options silently
- [ ] Tests for all combinations and edge cases

### LINKED ISSUE #190: feat(netcdf): add group navigation (group_names, get_group, group-qualified variable names)
## Context

NetCDF files with sub-groups (e.g. forecast models, nested hierarchies)
require navigation into sub-groups to access their variables.

## Problem / Current Behaviour

No API to list sub-groups, open a sub-group as a container, or access
variables via group-qualified paths like `"forecast/temperature"`.

## Proposed Solution

- `group_names` property: list sub-group names from root group
- `get_group(name)` method: open a sub-group as a new NetCDF container
  (creates MEM MDIM dataset, copies dimensions and arrays from the sub-group)
- Support nested paths via `/` separator (e.g. `"forecast/surface"`)
- `get_variable("forecast/temperature")` automatically navigates to the
  group and extracts the variable

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.group_names` | new property |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.get_group` | new method |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.get_variable` | group-qualified support |

## Effort Estimate

**Size**: `L`
**Rationale**: complex GDAL group-to-dataset conversion + test file (245 lines)

## Definition of Done

- [ ] `group_names` returns sub-group names
- [ ] `get_group()` returns a functional NetCDF container with the sub-group's
      variables and dimensions
- [ ] Nested group paths work (`"forecast/surface"`)
- [ ] `get_variable("group/var")` delegates to `get_group` then `get_variable`
- [ ] Tests cover single-level and nested groups

### LINKED ISSUE #191: feat(netcdf): add global attribute read/write/delete
## Context

Users need to read, set, and delete global attributes (e.g. `Conventions`,
`history`, `title`) on NetCDF files for CF compliance and metadata management.

## Problem / Current Behaviour

No direct API for global attribute access. Users must drop to raw GDAL calls.

## Proposed Solution

- `global_attributes` property: read-only dict from root group attributes
  (MDIM mode) or `GetMetadata()` (classic mode)
- `set_global_attribute(name, value)`: create or update a single attribute
  (handles str, int, float types; deletes existing attr first to avoid GDAL
  duplicate error)
- `delete_global_attribute(name)`: remove an attribute from root group

Both mutating methods call `_invalidate_caches()` after modification.

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.global_attributes` | new property |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.set_global_attribute` | new method |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.delete_global_attribute` | new method |

## Effort Estimate

**Size**: `S`
**Rationale**: three methods + test file (218 lines)

## Definition of Done

- [ ] `global_attributes` returns dict of root group attributes
- [ ] `set_global_attribute` creates/updates attributes
- [ ] `delete_global_attribute` removes attributes
- [ ] Cache invalidation on mutation
- [ ] Tests for read, write, update, delete, and type handling

### LINKED ISSUE #192: feat(netcdf): add convenience methods crop_variable, reproject_variable, resample_variable
## Context

The pattern `get_variable → operation → set_variable` is repeated frequently.
Chainable convenience methods reduce boilerplate.

## Problem / Current Behaviour

```python
var = nc.get_variable("temperature")
cropped = var.crop(mask)
nc.set_variable("temperature", cropped)
```

## Proposed Solution

Add chainable methods that combine get→operation→set:
- `crop_variable(name, mask, touch=True)` → returns self
- `reproject_variable(name, to_epsg, method)` → returns self (materializes VRT)
- `resample_variable(name, cell_size, method)` → returns self

All return `self` for chaining: `nc.crop_variable(...).resample_variable(...)`

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.crop_variable` | new |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.reproject_variable` | new |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.resample_variable` | new |

## Effort Estimate

**Size**: `S`
**Rationale**: three short methods + test file (340 lines)

## Definition of Done

- [ ] All three methods work and return self
- [ ] reproject_variable materializes VRT to avoid dangling refs
- [ ] Chaining works: `nc.crop_variable(...).resample_variable(...)`
- [ ] Band count and metadata preserved through operations

### LINKED ISSUE #193: feat(netcdf): add rename_variable method
## Context

Users need to rename variables in NetCDF containers (e.g. to match CF
conventions or standardize across datasets).

## Problem / Current Behaviour

No direct rename API. Users must manually extract data, create a new
variable, and delete the old one.

## Proposed Solution

Add `rename_variable(old_name, new_name)` that:
1. Validates old_name exists and new_name doesn't
2. Opens the MDArray via root group
3. Copies it under the new name via `_add_md_array_to_group`
4. Deletes the old MDArray
5. Invalidates caches

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.rename_variable` | new method |

## Effort Estimate

**Size**: `S`
**Rationale**: one method + test file (233 lines)

## Definition of Done

- [ ] Rename succeeds and variable accessible under new name
- [ ] Old name no longer in variable_names
- [ ] Data preserved after rename
- [ ] Raises ValueError for nonexistent old_name or existing new_name
- [ ] Tests cover happy path, errors, and data integrity

### LINKED ISSUE #194: fix(netcdf): Y-flip consistency in _read_variable and _needs_y_flip refactor
## Context

`get_variable()` flips the Y axis for south-to-north data, but
`_read_variable()` did not, causing inconsistent array orientation
between the two code paths.

## Problem / Current Behaviour

- `_needs_y_flip` was a static method checking a classic dataset's
  geotransform, only used in `get_variable`
- `_read_variable` (used for reading dimension variables and raw data)
  did not apply the Y flip, leading to upside-down data when accessed
  directly
- `plot()` not blocked on root MDIM containers

## Proposed Solution

- Refactor `_needs_y_flip` to an instance method that checks the MDArray's
  Y dimension directly via `AsClassicDataset` geotransform. Returns False
  for 1-D arrays.
- Apply Y-flip in `_read_variable` for arrays with 2+ dimensions when
  `_needs_y_flip` returns True (matching `get_variable` behavior)
- Block `plot()` on root MDIM containers via `_check_not_container`

Also includes `_dtype_to_str` improvement in utils.py: uses
`gdal_to_numpy_dtype()` for numeric types instead of falling back to
`str(dt)` which produced unhelpful output.

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF._needs_y_flip` | refactored |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF._read_variable` | add Y-flip |
| `src/pyramids/netcdf/netcdf.py` | `NetCDF.plot` | block on container |
| `src/pyramids/netcdf/utils.py` | `_dtype_to_str` | use gdal_to_numpy_dtype |

## Effort Estimate

**Size**: `M`
**Rationale**: refactor + consistency fix + new test file (197 lines)

## Definition of Done

- [ ] `_read_variable` produces same orientation as `get_variable`
- [ ] `_needs_y_flip` works on MDArray directly
- [ ] `plot()` raises ValueError on root containers
- [ ] `_dtype_to_str` returns numpy dtype names for numeric types
- [ ] test_y_orientation.py covers all orientation scenarios

### LINKED ISSUE #195: fix(netcdf): _add_md_array_to_group no longer falls back to -9999 nodata
## Context

When copying MDArrays between groups, the code silently injected `-9999`
as the nodata value when the source array's `GetNoDataValue()` failed,
masking the real issue and corrupting metadata.

## Problem / Current Behaviour

```python
try:
    new_md_array.SetNoDataValueDouble(src_mdarray.GetNoDataValue())
except Exception:
    new_md_array.SetNoDataValueDouble(-9999)  # silent fallback
```

## Proposed Solution

Only set nodata if the source array actually has one (`GetNoDataValue()`
returns non-None). If it fails, skip rather than inject a fake value.

```python
ndv = src_mdarray.GetNoDataValue()
if ndv is not None:
    try:
        new_md_array.SetNoDataValueDouble(ndv)
    except Exception:
        pass
```

### Affected locations

| File | Symbol | Notes |
|------|--------|-------|
| `src/pyramids/netcdf/netcdf.py` | `NetCDF._add_md_array_to_group` | fix fallback |

## Effort Estimate

**Size**: `XS`
**Rationale**: 4 lines changed

## Definition of Done

- [ ] No silent -9999 injection
- [ ] Nodata only set when source has one
- [ ] Existing tests pass

### PR DESCRIPTION
**Pull request by MAfarrag**
## Summary

Major refinement of the NetCDF metadata and NetCDF class APIs:
- Rename ArrayInfo → VariableInfo across the codebase
- Add sel(), chunking/compression, group navigation, global attributes,
  convenience methods, rename_variable, and unpack support
- Enable crop/to_crs/resample on root MDIM containers
- Lazy-load variables, fix Y-flip consistency, fix silent -9999 nodata fallback
- Overhaul test suite: 2545 lines added, 1162 removed across 16 test files

## Issues

- Fixes #182
- Fixes #183
- Fixes #184
- Fixes #185
- Fixes #186
- Fixes #187
- Fixes #188
- Fixes #189
- Fixes #190
- Fixes #191
- Fixes #192
- Fixes #193
- Fixes #194
- Fixes #195

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Breaking Changes

- `ArrayInfo` renamed to `VariableInfo`
- `NetCDFMetadata.arrays` renamed to `NetCDFMetadata.variables`
- `GroupInfo.arrays` renamed to `GroupInfo.variables`
- `NetCDFMetadata.names` property removed (use `list(meta.dimensions.keys())`)
- `NetCDFMetadata.dimension_overview` field removed
- Serialization keys changed from `arrays` to `variables` in JSON output
- Dimension/variable/group dictionary keys no longer have leading `/`

## How Has This Been Tested?

- 16 test files modified/added covering all new features and refactors
- Tests run across ubuntu/windows/macos x Python 3.11/3.12/3.13 via CI

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Documentation is updated

### CODE REVIEW

File: src/pyramids/dataset/ops/vectorize.py [hunk 1]
```diff
@@ -187,61 +192,93 @@ def to_feature_collection(
...
-                    # Handle single band rasters
-                    idx = (0, 1)
-
-                mask_arr = np.ones((arr.shape[idx[0]], arr.shape[idx[1]]))
-                pixels = get_pixels(arr, mask_arr).transpose()
-                df_list.append(pd.DataFrame(pixels, columns=band_names))
-
-            # Merge all the tiles.
-            df = pd.concat(df_list)
+            df = self._extract_values_tiled(band_names, tile_size)
```
[the remark this answers is not in the document]
Comment (MAfarrag) [reply src/pyramids/dataset/ops/vectorize.py]: solved

File: src/pyramids/dataset/ops/vectorize.py [hunk 2]
```diff
@@ -187,61 +192,93 @@ def to_feature_collection(
...
+        """
+        df_list = []
+        for arr in self.get_tile(tile_size):
+            idx = (1, 2) if arr.ndim > 2 else (0, 1)
+            mask_arr = np.ones((arr.shape[idx[0]], arr.shape[idx[1]]))
+            pixels = get_pixels(arr, mask_arr).transpose()
+            df_list.append(pd.DataFrame(pixels, columns=band_names))
+
+        return pd.concat(df_list)
+
```
[the remark this answers is not in the document]
Comment (MAfarrag) [reply src/pyramids/dataset/ops/vectorize.py]: solved

File: src/pyramids/dataset/ops/band_metadata.py [hunk 3]
```diff
@@ -863,7 +863,8 @@ def _check_no_data_value(self, no_data_value: list):
                 )
                 no_data_value[i] = self.numpy_dtype[i](DEFAULT_NO_DATA_VALUE)
         return no_data_value
```
[the remark this answers is not in the document]
Comment (MAfarrag) [reply src/pyramids/dataset/ops/band_metadata.py]: solved
